lib/matplotlib/rcsetup.py

```diff
@@ -751,51 +728,6 @@ def visit_Attribute(self, node):
         self.generic_visit(node)
 
 
-# A validator dedicated to the named legend loc
-_validate_named_legend_loc = ValidateInStrings(
-    'legend.loc',
-    [
-        "best",
-        "upper right", "upper left", "lower left", "lower right", "right",
-        "center left", "center right", "lower center", "upper center",
-        "center"],
-    ignorecase=True)
-
-
-def _validate_legend_loc(loc):
-    """
-    Confirm that loc is a type which rc.Params["legend.loc"] supports.
-
-    .. versionadded:: 3.8
-
-    Parameters
-    ----------
-    loc : str | int | (float, float) | str((float, float))
-        The location of the legend.
-
-    Returns
-    -------
-    loc : str | int | (float, float) or raise ValueError exception
-        The location of the legend.
-    """
-    if isinstance(loc, str):
-        try:
-            return _validate_named_legend_loc(loc)
-        except ValueError:
-            pass
-        try:
-            loc = ast.literal_eval(loc)
-        except (SyntaxError, ValueError):
-            pass
-    if isinstance(loc, int):
-        if 0 <= loc <= 10:
-            return loc
-    if isinstance(loc, tuple):
-        if len(loc) == 2 and all(isinstance(e, Real) for e in loc):
-            return loc
-    raise ValueError(f"{loc} is not a valid legend location.")
-
-
```

Comment:
Why were these removed?

`legend_loc` also accepts non stringed entries, such as `(0.8, 0.8)` So the replacement which only accepts the strings is incorrect.